lib/puppet-lint/plugins/check_classes.rb

```diff
@@ -14,6 +14,44 @@ def check
   end
 end
 
+# Public: Test the manifest tokens for chaining arrow that is
+# on the line of the left operand when the right operand is on another line.
+#
+# https://docs.puppet.com/guides/style_guide.html#chaining-arrow-syntax
+PuppetLint.new_check(:arrow_on_right_operand_line) do
+  def check
+    tokens.select { |r| r.type == :IN_EDGE }.each do |token|
```

Comment:
This should probably apply to the subscribe chain operator (`~>`/`:IN_EDGE_SUB`) too, I think.